js-packages/@prestojs/rest/src/Endpoint.ts

```diff
@@ -511,6 +554,38 @@ function isEqualPrepareKey(a: ExecuteInitOptions, b: ExecuteInitOptions): boolea
  * new Endpoint('/users/', { middleware: [csrfTokenMiddleware] })
  * ```
  *
+ * When middleware is passed to the `Endpoint` it is _appended_ by to the default
+ * middleware specified in `Endpoint.defaultConfig.middleware`.
+ *
+ * To change how middleware is combined per `Endpoint` you can specify the
+ * `getMiddleware` option. This is passed the middleware for the `Endpoint` and
+ * the `Endpoint` itself and should return an array of middleware to use.
+ *
+ * The default implementation looks like
+ *
+ * ```js
+ * (middleware) => [
+ *   requestDefaultsMiddleware,
+ *   ...Endpoint.defaultConfig.middleware,
+ *   ...middleware,
+ * ]
+ * ```
+ *
+ * The [requestDefaultsMiddleware](doc:requestDefaultsMiddleware) is included here
+ * rather than in `Endpoint.defaultConfig.middleware` as it's strongly recommended
+ * to have this on and would be easy to inadvertently remove it from the global
+ * middleware.
```

Comment:
I think that make it worse

You now have to remember to include `requestDefaultsMiddleware` every time you override `getMiddleware()` instead of having to remember to put it once in the global middleware.

Using django as an example: there's a bunch of middleware that's there in every project, you're always starting from a default list instead of creating it from scratch anyway

Reply:
ok will move it

not convinced having this middleware was the best idea given other middleware relies on it but i guess it gives ultimate flexibility

Reply:
i've added to the docs the recommendation to add to the global middleware rather than replacing it